structuredheadings/plugin.js

```diff
@@ -31,12 +38,39 @@
         }
       };
 
+      var getCurrentBlock = function (element) {
+        return element.getAscendant({
```

Comment:
it looks like they might be slightly more robust (iterate through the CKEDITOR dtd and add relevant block elements)